pkg/sdk/testint/tasks_gen_integration_test.go

```diff
@@ -592,4 +592,135 @@ func TestInt_Tasks(t *testing.T) {
+
+		require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(middleTask.ID()).WithResume(sdk.Bool(true))))
+		t.Cleanup(func() {
+			require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(middleTask.ID()).WithSuspend(sdk.Bool(true))))
+		})
+
+		require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(rootTask.ID()).WithResume(sdk.Bool(true))))
+		t.Cleanup(func() {
+			require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(rootTask.ID()).WithSuspend(sdk.Bool(true))))
+		})
+
+		tasksToResume, err := client.Tasks.SuspendRootTasks(ctx, task.ID(), task.ID())
+		require.NoError(t, err)
+		require.NotEmpty(t, tasksToResume)
+		require.Contains(t, tasksToResume, rootTask.ID())
+
+		rootTaskStatus, err := client.Tasks.ShowByID(ctx, rootTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateSuspended, rootTaskStatus.State)
+
+		middleTaskStatus, err := client.Tasks.ShowByID(ctx, middleTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateStarted, middleTaskStatus.State)
+
+		require.NoError(t, client.Tasks.ResumeTasks(ctx, tasksToResume))
+
+		rootTaskStatus, err = client.Tasks.ShowByID(ctx, rootTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateStarted, rootTaskStatus.State)
+
+		middleTaskStatus, err = client.Tasks.ShowByID(ctx, middleTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateStarted, middleTaskStatus.State)
+	})
+
+	t.Run("suspend root tasks temporarily with three sequentially connected tasks - middle in DAG", func(t *testing.T) {
+		rootTaskId := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		rootTask := createTaskWithRequest(t, sdk.NewCreateTaskRequest(rootTaskId, sql).WithSchedule(sdk.String("60 minutes")))
+
+		middleTaskId := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		middleTask := createTaskWithRequest(t, sdk.NewCreateTaskRequest(middleTaskId, sql).WithAfter([]sdk.SchemaObjectIdentifier{rootTask.ID()}))
+
+		childTaskId := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		childTask := createTaskWithRequest(t, sdk.NewCreateTaskRequest(childTaskId, sql).WithAfter([]sdk.SchemaObjectIdentifier{middleTask.ID()}))
+
+		require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(childTask.ID()).WithResume(sdk.Bool(true))))
+		t.Cleanup(func() {
+			require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(childTask.ID()).WithSuspend(sdk.Bool(true))))
+		})
+
+		require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(rootTask.ID()).WithResume(sdk.Bool(true))))
+		t.Cleanup(func() {
+			require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(rootTask.ID()).WithSuspend(sdk.Bool(true))))
+		})
+
+		tasksToResume, err := client.Tasks.SuspendRootTasks(ctx, middleTask.ID(), middleTask.ID())
```

Comment:
is this invocation right? I guess we are invoking this function with two different task ids and that's why we need both?

**EDIT**: okay, I see that for update we invoke it with the same id... so yes, I think that what you wrote about checking the necessity for two of them is valid in a follow-up task.

Reply:
Yeah, It's confusing that sometimes it was with the same id and sometimes we need to pass two different ones. For now, I implemented it in a way it was working before, but in the follow-up task I created, this should be analyzed (to answer questions like why we need 2 parameters? If not, can we remove one of them and stick with only one?).